### PR TITLE
Remove CMake version checks

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Now that our minimum CMake version is 3.20, we can remove these version checks.

Contributes to https://github.com/dotnet/runtime/issues/88688

### CODE REVIEW

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

File: src/tests/CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,7 @@
 cmake_minimum_required(VERSION 3.20)
 
 cmake_policy(SET CMP0042 NEW)
-if (CMAKE_VERSION VERSION_GREATER 3.15 OR CMAKE_VERSION VERSION_EQUAL 3.15)
-  cmake_policy(SET CMP0091 NEW)
-endif()
+cmake_policy(SET CMP0091 NEW)
```
Comment (jkotas) [inline src/tests/CMakeLists.txt:4]: Do we need to set this explicitly? 

`cmake_mimum_required(VERSION 3.20)` should set these policies to NEW by default.

Comment (jkoritzinsky) [reply src/tests/CMakeLists.txt:4]: I can check. I know that some of these policies need to be explicitly set to the `NEW` behavior. I believe this policy is one of them based on the docs.

Comment (jkotas) [reply src/tests/CMakeLists.txt:4]: https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html says that all policies are set to `NEW` behavior based on min version, it does not mention any exceptions.

> The cmake_minimum_required(VERSION) command implicitly invokes the [cmake_policy(VERSION)](https://cmake.org/cmake/help/latest/command/cmake_policy.html#command:cmake_policy) command to specify that the current project code is written for the given range of CMake versions. All policies known to the running version of CMake and introduced in the `<min>` (or `<max>`, if specified) version or earlier will be set to use NEW behavior.

Comment (jkoritzinsky) [reply src/tests/CMakeLists.txt:4]: The documentation for this policy states the following:

> This policy was introduced in CMake version 3.15. Use the cmake_policy() command to set it to OLD or NEW explicitly. Unlike many policies, CMake version 3.27.4 does not warn when this policy is not set and simply uses OLD behavior.

Comment (jkoritzinsky) [reply src/tests/CMakeLists.txt:4]: I'll try removing it and see what happens.

**Review by jkotas — APPROVED** [review]

**Pull request merged**